codex/contracts/interactions/hostinteractions.nim

```diff
@@ -0,0 +1,62 @@
+import pkg/ethers
+import pkg/chronicles
+import pkg/questionable
+import pkg/questionable/results
+
+import ../../sales
+import ../../proving
+import ../../stores
+import ../deployment
+import ../proofs
+import ./interactions
+
+export sales
+export proving
+export chronicles
+
+type
+  HostInteractions* = ref object of ContractInteractions
+    sales*: Sales
+    proving*: Proving
+
+proc new*(_: type HostInteractions,
+          signer: Signer,
+          deployment: Deployment,
```

Comment:
Same as for ClientInteractions, I'd avoid using `Deployment` here, and perhaps use `marketplaceAddress` instead.